js/components/addressBook/addressBook_controller.js

```diff
@@ -4,8 +4,22 @@ angular.module('contactsApp')
 
 	ctrl.showUrl = false;
 	/* globals oc_config */
+
+	function compareVersion(version1, version2) {
+		var result=false;
+		for(var i=0;i<(Math.max(version1.length, version2.length));i++) {
```

Comment:
Yes we do.
If we're comparing 9.0.0 and 9.0.0.1 we still need to check the last value. But I need to add a safety conversion if NaN or undefined.